sentry-android-replay/src/main/java/io/sentry/android/replay/capture/BaseCaptureStrategy.kt

```diff
@@ -295,48 +295,110 @@ internal abstract class BaseCaptureStrategy(
+                                    pos
+                                }
+                                this.pointerId = pointerId
+                            }
+                            currentPositions[pointerId]!!.clear()
                         }
-                    }.also {
-                        currentPositions.clear()
-                        touchMoveBaseline = 0L
                     }
+                    touchMoveBaseline = 0L
+                    moveEvents
                 } else {
                     null
                 }
             }
 
-            MotionEvent.ACTION_DOWN, MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
-                RRWebInteractionEvent().apply {
-                    timestamp = dateProvider.currentTimeMillis
-                    x = event.x * recorderConfig.scaleFactorX
-                    y = event.y * recorderConfig.scaleFactorY
-                    id = 0 // html node id, but we don't have it, so hardcode to 0 to align with FE
-                    interactionType = when (action) {
-                        MotionEvent.ACTION_UP -> InteractionType.TouchEnd
-                        MotionEvent.ACTION_DOWN -> InteractionType.TouchStart
-                        MotionEvent.ACTION_CANCEL -> InteractionType.TouchCancel
-                        else -> InteractionType.TouchMove_Departed // should not happen
+            MotionEvent.ACTION_DOWN,
+            MotionEvent.ACTION_POINTER_DOWN -> {
+                val pId = event.getPointerId(event.actionIndex)
+                val pIndex = event.findPointerIndex(pId)
+
+                if (pIndex == -1) {
+                    // no data for this pointer
+                    return null
+                }
+
+                // new finger down - add a new pointer for tracking movement
+                currentPositions[pId] = ArrayList()
+                listOf(
+                    RRWebInteractionEvent().apply {
+                        timestamp = dateProvider.currentTimeMillis
+                        x = event.getX(pIndex) * recorderConfig.scaleFactorX
+                        y = event.getY(pIndex) * recorderConfig.scaleFactorY
+                        id = 0 // html node id, but we don't have it, so hardcode to 0 to align with FE
+                        pointerId = pId
+                        interactionType = InteractionType.TouchStart
                     }
+                )
+            }
+            MotionEvent.ACTION_UP,
+            MotionEvent.ACTION_POINTER_UP -> {
+                val pId = event.getPointerId(event.actionIndex)
+                val pIndex = event.findPointerIndex(pId)
+
+                if (pIndex == -1) {
+                    // no data for this pointer
+                    return null
                 }
+
+                // finger lift up - remove the pointer from tracking
+                currentPositions[pId]?.clear()
+                currentPositions.remove(pId)
```

Comment:
wouldn't it be enough to just remove the `pid` entry?
```suggestion
                currentPositions.remove(pId)
```